### PR TITLE
Fix minor issues like NULL pointer check 

### PR DESCRIPTION
**Pull request by ajinkyaraj-23**
- **Add assert conditions while accessing baking_key in apdu_query.c**
- **Refactor: change magic numbers to constant macros for better readability.**
- **Refactor: Fix or Remove all TODO's**
- **Fix null pointer checks as pre-conditions for functions in to_string.c**
- **Fix issues with pytest environment.**

### CODE REVIEW

File: src/apdu_sign.c [hunk 1]
```diff
@@ -375,7 +375,10 @@ static int perform_signature(bool const send_hash) {
 
     TZ_ASSERT(os_global_pin_is_validated() == BOLOS_UX_OK, EXC_SECURITY);
 
-    TZ_ASSERT(write_high_water_mark(&G.parsed_baking_data), EXC_MEMORY_ERROR);
+    exc = write_high_water_mark(&G.parsed_baking_data);
+    if (exc != SW_OK) {
+        goto end;
+    }
```
Comment (spalmer25) [inline src/apdu_sign.c:381]:
```suggestion
TZ_CHECK(write_high_water_mark(&G.parsed_baking_data));
```

Comment (ajinkyaraj-23) [reply src/apdu_sign.c]: Done.

File: README.md [hunk 2]
```diff
@@ -142,26 +142,23 @@ BOLOS_SDK=$NANOS_SDK make DEBUG=1
...
+The application tests are run using same docker container used for building. Inside the docker container run following script,
 ```
-$ sudo apt-get update && sudo apt-get install -y qemu-user-static tesseract-ocr libtesseract-dev python3-pip libgmp-dev libsodium-dev git
-$ python3 -m venv env
-$ source env/bin/activate
-```
-Inside the virtualenv, load the requirements.txt file.
-```
-(env)$ pip install -r test/requirements.txt
+$ apk add gmp-dev curl jq libsodium-dev git xxd procps;
```
Comment (spalmer25) [inline README.md:147]: `curl`, `jq`, `git`, `xxd` and `procps` and not required

Comment (ajinkyaraj-23) [reply README.md]: Removed.

**Review by spalmer25 — APPROVED** [review]

**Pull request merged**